internal/network/command_gateway_describe.go

```diff
@@ -0,0 +1,89 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/output"
+)
+
+func (c *command) newGatewayDescribeCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "describe <id>",
+		Short:             "Describe a gateway.",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validGatewayArgs),
+		RunE:              c.gatewayDescribe,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Describe gateway "gw-123456".`,
+				Code: "confluent network gateway describe gw-123456",
+			},
+		),
+	}
+
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *command) gatewayDescribe(cmd *cobra.Command, args []string) error {
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	gateway, err := c.V2Client.GetGateway(environmentId, args[0])
+	if err != nil {
+		return err
+	}
+
+	if gateway.Spec == nil {
+		return fmt.Errorf(errors.CorruptedNetworkResponseErrorMsg, "spec")
+	}
+	if gateway.Status == nil {
+		return fmt.Errorf(errors.CorruptedNetworkResponseErrorMsg, "status")
+	}
+
+	out := &gatewayOut{
+		Id:          gateway.GetId(),
+		Name:        gateway.Spec.GetDisplayName(),
+		Environment: gateway.Spec.Environment.GetId(),
+		Phase:       gateway.Status.GetPhase(),
+	}
+
+	if gateway.Spec.Config != nil && gateway.Spec.Config.NetworkingV1AwsEgressPrivateLinkGatewaySpec != nil {
+		out.Region = gateway.Spec.Config.NetworkingV1AwsEgressPrivateLinkGatewaySpec.GetRegion()
+	}
+	if gateway.Spec.Config != nil && gateway.Spec.Config.NetworkingV1AwsPeeringGatewaySpec != nil {
+		out.Region = gateway.Spec.Config.NetworkingV1AwsPeeringGatewaySpec.GetRegion()
+	}
+	if gateway.Spec.Config != nil && gateway.Spec.Config.NetworkingV1AzureEgressPrivateLinkGatewaySpec != nil {
+		out.Region = gateway.Spec.Config.NetworkingV1AzureEgressPrivateLinkGatewaySpec.GetRegion()
+	}
+	if gateway.Spec.Config != nil && gateway.Spec.Config.NetworkingV1AzurePeeringGatewaySpec != nil {
+		out.Region = gateway.Spec.Config.NetworkingV1AzurePeeringGatewaySpec.GetRegion()
+	}
+
+	cloud, err := getGatewayCloud(gateway)
+	if err != nil {
+		return err
+	}
+
+	switch cloud {
+	case CloudAws:
+		out.AwsPrincipalArn = gateway.Status.CloudGateway.NetworkingV1AwsEgressPrivateLinkGatewayStatus.GetPrincipalArn()
+	case CloudAzure:
+		out.AzureSubscription = gateway.Status.CloudGateway.NetworkingV1AzureEgressPrivateLinkGatewayStatus.GetSubscription()
```

Comment:
I'm unsure of this part. I assumed that one of these must be non-nil, but is that still true if it's one of the peering gateways?

Reply:
Peering gateways won’t have either of these. Only AwsEgressPrivateLink gateways will have a `principal_arn`, and AzureEgressPrivateLink gateways for `subscription`.

Reply:
I think we also need to update the output for the network describe/list commands to include `gateway` so customers can grab the `gateway id`.

@ay-zhang can you confirm?

Reply:
this was added previously! https://github.com/confluentinc/cli/pull/2599

Reply:
Oh sweet. Thanks!

Reply:
@syoung-cflt It looks like the `network describe | list` commands already include the gateway ID in the output.